src/System.Management.Automation/engine/TypeTable.cs

```diff
@@ -2191,7 +2191,7 @@ public TypeMemberData StringSerializationSourceProperty
                     return;
                 }
 
-                if (!(value is NotePropertyData || value is ScriptPropertyData || value is CodePropertyData))
+                if (value is NotePropertyData || value is ScriptPropertyData || value is not CodePropertyData)
```

Comment:
This will break the original logic. I'd suggest:
```suggestion
                if (value is not NotePropertyData && value is not ScriptPropertyData && value is not CodePropertyData)
```